src/services/nettotals.services.ts

```diff
@@ -0,0 +1,73 @@
+import { Between, FindManyOptions, getRepository, Repository } from 'typeorm';
+
+import { NettotalsEntity } from '../entity/nettotals.entity';
+import { getStartPoint, TPeriod } from '../utils/period';
+
+class StatsNetTotalsService {
+  private getRepository(): Repository<NettotalsEntity> {
+    return getRepository(NettotalsEntity);
+  }
+  async getLatest(): Promise<NettotalsEntity | null> {
+    const items = await this.getRepository().find({
+      order: { timestamp: 'DESC' },
+      take: 1,
+    });
+    return items.length === 1 ? items[0] : null;
+  }
+
+  async getAll(
+    offset: number,
+    limit: number,
+    orderBy: keyof NettotalsEntity,
+    orderDirection: 'DESC' | 'ASC',
+    period: TPeriod,
+  ) {
+    const query: FindManyOptions<NettotalsEntity> = {
+      // skip: offset,
+      // take: limit,
+      order: {
+        [orderBy]: orderDirection,
+      },
+    };
+    if (period) {
+      const fromTime = getStartPoint(period);
+      query.where = {
+        timestamp: Between(fromTime, new Date().getTime()),
+      };
+    }
+    if (offset) {
+      query.skip = offset;
+    }
+    if (limit) {
+      query.take = limit;
+    }
+    if (limit) {
+      const statsInfo = await this.getRepository().find(query);
+      return statsInfo;
+    }
+    const count = await this.getRepository().count(query);
+    const take = 500;
+    const skip = Math.round(count / take);
+    let data = [];
+    // get statistics data limit 500 for chart
+    if (count <= take || skip < 2) {
+      const statsInfo = await this.getRepository().find(query);
+      return statsInfo;
+    } else {
+      let index = 0;
+      for (let i = 0; i <= count; i += skip) {
+        const item = await this.getRepository().find({
```

Comment:
The same as above about requesting database in a loop

I see this code is repeated for blocks, nettotals, stats.mining, stats and rawempoolinfo.
Please, don't copy-paste code in such way, consider moving repeated code to common service

Reply:
Okay,  I moved it to the chartData.services.ts file